content/posts/2020/05-03-Tutorial-Combining-Nornir-&-pyATS.md

```diff
@@ -0,0 +1,146 @@
+---
+title: "Tutorial: Combining Nornir with pyATS"
+date: "2020-05-03"
+tags: ["nornir", "pyATS"]
+---
+
+This tutorial aims to provide a simple use-case for combining Nornir and pyATS together in order to profile your current network configurations and implement desired state - as specified in host variable definition files.
+
+The workflow we will follow in this tutorial will be first:
+- Deploy OSPF desired state using Nornir
+- Immediately deploying pyATS to capture a golden image profile of this &quot;desired state&quot;
+----------------------------------------------------------------------------------------------------
+
+After we have conducted the initial steps we will then deploy a python script (Pynir.py) that will:
+
+- Invoke pyATS to profile the current OSPF configuration and compare it to the golden image previously captured.
+- If pyATS detects a difference – the terminal will prompt an alert signalling that OSPF in currently out of sync with desired state
+- The script will then give the user the option to rollback to desired state
+- Should the user answer **No** – the script ends and all current OSPF settings and Diff artefacts are left and are able to be inspected
+- Should the user answer **Yes** – the script invokes Nornir to first erase all OSPF configurations currently in the network, before reimplementing desired state by pulling information from host variable definition files and leveraging jinja2 templating.
+- Lastly, if the script detects that the current OSPF profile is identical to the OSPF desired state, it will generate a message informing us that all current configurations are matching our desired state.
+
+
+
+To begin, let&#39;s first look at the directory structure and setup:
+
+![alt text](https://github.com/IPvZero/Nornir-Blog/blob/master/images/1.png?raw=true)
```

Comment:
would you mind copying the images to this repo to avoid broken images if something changes in the other repo in the future?

You can add them under `/static/img/posts/$NAME-OF-ENTRY/...` and you should be able to reference them with the same path. Also, set the `alt text`

Same for the rest of the images

Reply:
Actually, looking at the content of the pictures, maybe it's better to just copy and paste the code